x/protosink/protosink.go

```diff
@@ -0,0 +1,97 @@
+// Package protosink provides a cache.Sink using Protobuf
+// encoding
+package protosink
+
+import (
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/darvaza-proxy/cache"
+)
+
+var (
+	_ cache.Sink = (*ProtoSink)(nil)
+)
+
+// ProtoSink is a Sink using Proto for encoding
+type ProtoSink struct {
+	out proto.Message
+	b   []byte
+	e   time.Time
+}
+
+// Bytes returns the Protobuf encoded representation of the
+// object in the Sink.
+func (sink *ProtoSink) Bytes() []byte {
+	return sink.b
+}
+
+// Len returns the length of the Protobuf encoded representation of the
+// object in the Sink. 0 if empty
+func (sink *ProtoSink) Len() int {
+	return len(sink.b)
+}
+
+// Expire tells when this object will be evicted from the Cache
+func (sink *ProtoSink) Expire() time.Time {
+	return sink.e
+}
+
+// Reset clears everything but the type pointer assigned during creation
+func (sink *ProtoSink) Reset() {
+	sink.b = []byte{}
+	sink.e = time.Time{}
+	// TODO: how do we clear *sink.out ?
```

Comment:
since an interface zero value is `nil`  I would assume `sink.out=nil` would work, an interface accepts to be `nil` but what happens after that 
https://go.dev/play/p/nEiJVQmP9IB

Reply:
In your case it is a interface which aliases another interface, there is no struct to zero its fields so I would say that at most you can `nil` it, but I've been known to be wrong sometimes.